leku/src/main/java/com/schibstedspain/leku/timezone/TimeZoneUtils.java

```diff
@@ -0,0 +1,49 @@
+package com.schibstedspain.leku.timezone;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import com.google.maps.GeoApiContext;
+import com.google.maps.TimeZoneApi;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.LatLng;
+
+import java.io.IOException;
+import java.util.TimeZone;
+
+public class TimeZoneUtils {
+  public static TimeZone getTimeZone(Context context, double lng, double lat) {
+    String api_key = getGoogleMetadata(context);
+    if (api_key == null || api_key.length() <= 0) {
+      Log.d(TimeZoneUtils.class.getName(), "No Google Maps api specified");
+      return null;
+    }
+
+    GeoApiContext geoContext = new GeoApiContext().setApiKey(api_key);
+    try {
+      return TimeZoneApi.getTimeZone(geoContext, new LatLng(lng, lat)).await();
```

Comment:
That's why I told you to do it asynchronously, despite of doing it in other place, basically to not blocking the view (performance).